stl/inc/variant

```diff
@@ -1447,14 +1453,39 @@ _NODISCARD constexpr bool operator>=(const variant<_Types...>& _Left, const vari
     conjunction_v<is_nothrow_invocable_r<bool, greater_equal<>, const _Types&, const _Types&>...>) /* strengthened */ {
     // determine if _Left's index() is greater than _Right's, or equal and
     // _Left contains a value greater than or equal to _Right
-    using _Visitor             = _Variant_relop_visitor<greater_equal<>, _Types...>;
+    using _Visitor             = _Variant_relop_visitor<greater_equal<>, bool, _Types...>;
     const size_t _Left_offset  = _Left.index() + 1;
     const size_t _Right_offset = _Right.index() + 1;
     return _Left_offset > _Right_offset
         || (_Left_offset == _Right_offset
             && _Variant_raw_visit(_Right_offset - 1, _Right._Storage(), _Visitor{_Left._Storage()}));
 }
 
+#ifdef _HAS_CXX20
+struct _Variant_relop_3way_comparison {
+    template <class _Ty>
+    compare_three_way_result_t<_Ty> operator()(const _Ty& _Left, const _Ty& _Right) {

```

Comment:
This should be a `const` member. Conventionally, we aggressively mark "pure" functions - even internal ones - `_NODISCARD` to ensure they're not misused; this looks like a good candidate.

Could we use `std::compare_three_way` instead of this custom function object? I suppose we want to avoid reevaluating the constraint which is effectively enforced with return type SFINAE on 1473.

Reply:
This was just a case of not knowing about this free function rather than trying to avoid reevaluating the constraint. I don't think that warrants a custom function object and will change this over.